addons/mail/static/tests/chatter_tests.js

```diff
@@ -839,6 +839,61 @@ QUnit.test('chatter: post a message disable the send button', function(assert) {
     form.destroy();
 });
 
+QUnit.test('chatter: post message failure keep message', function(assert) {
+    assert.expect(4);
+    var form = createView({
+        View: FormView,
+        model: 'partner',
+        data: this.data,
+        services: this.services,
+        arch: '<form string="Partners">' +
+                '<sheet>' +
+                    '<field name="foo"/>' +
+                '</sheet>' +
+                '<div class="oe_chatter">' +
+                    '<field name="message_ids" widget="mail_thread"/>' +
+                '</div>' +
+            '</form>',
+        res_id: 2,
+        session: {},
+        mockRPC: function (route, args) {
+            if (args.method === 'message_get_suggested_recipients') {
+                return $.when({2: []});
+            }
+            if (args.method === 'message_post') {
+                assert.ok(form.$('.o_composer_button_send').prop("disabled"),
+                    "Send button should be disabled when a message is being sent");
+                return $.Deferred().reject();
+            }
+            if (args.method === 'message_format') {
+                return $.when([{
+                    author_id: ["42", "Me"],
+                    model: 'partner',
+                }]);
+            }
+            return this._super(route, args);
+        },
+        intercepts: {
```

Comment:
you can remove this

(this was necessary when the mail service was not available in tests, which is no longer the case in v11.2+)

Reply:
ok, I have also removed the one from the test case over this one